internal/kubeclient/manifest/manifest.go

```diff
@@ -1,73 +0,0 @@
-package manifest
-
-import (
-	"sync"
-
-	"github.com/pkg/errors"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/serializer"
-	"sigs.k8s.io/kustomize/api/provider"
-	"sigs.k8s.io/kustomize/api/resmap"
-	"sigs.k8s.io/kustomize/api/resource"
-	"sigs.k8s.io/kustomize/kyaml/openapi"
-)
-
-var (
-	initSchemaOnce = &sync.Once{}
-)
-
-type ParseFunc func(data []byte) (cluster, namespaced []*unstructured.Unstructured, err error)
-
-// NewParser returns a new parser that parses Kubernetes manifest and
-// returns parsed objects in cluster - namespaced order.
-func NewParser(scheme *runtime.Scheme) ParseFunc {
-	initSchemaOnce.Do(func() {
-		_ = openapi.Schema()
-	})
-	codecs := serializer.NewCodecFactory(scheme)
-	deserializer := codecs.UniversalDeserializer()
-	factory := resmap.NewFactory(provider.NewDefaultDepProvider().GetResourceFactory())
-
-	return func(data []byte) (cluster, namespaced []*unstructured.Unstructured, err error) {
-		resMap, err := factory.NewResMapFromBytes(data)
-		if err != nil {
-			return nil, nil, errors.Wrap(err, "new resmap from data")
-		}
-		cluster = make([]*unstructured.Unstructured, 0, resMap.Size())
-		for _, r := range resMap.ClusterScoped() {
```

Comment:
To make this recognize Project as a cluster-scoped resource required jumping through a lot of hoops that were more easily avoided by refactoring this whole parser into the new `splitYAML()` function that was adapted from GitOps Engine.